From 01803979df665c1152a887553217e5acc3ae34ae Mon Sep 17 00:00:00 2001
From: xzhang <xzhang@xzdt>
Date: Thu, 21 Jan 2016 10:32:53 -0800
Subject: [PATCH 0469/1164] CDH-34627: HIVE-12811: Name yarn application name
 more meaning than just 'Hive on Spark' (Rui
 reviewed by Xuefu)

Change-Id: I6bad3d9e38150cf22d52d983abfaa7694f7254b3
---
 .../java/org/apache/hadoop/hive/conf/HiveConf.java |    3 ++-
 .../hive/ql/exec/spark/HiveSparkClientFactory.java |    7 ++++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java b/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
index 652db5c..d4fbb65 100644
--- a/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
+++ b/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
@@ -2369,7 +2369,8 @@ public void verifyAndSet(String name, String value) throws IllegalArgumentExcept
   private boolean isSparkRelatedConfig(String name) {
     boolean result = false;
     if (name.startsWith("spark")) { // Spark property.
-      result = true;
+      // for now we don't support changing spark app name on the fly
+      result = !name.equals("spark.app.name");
     } else if (name.startsWith("yarn")) { // YARN property in Spark on YARN mode.
       String sparkMaster = get("spark.master");
       if (sparkMaster != null &&
diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
index 33dcb35..e02caa7 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
@@ -70,7 +70,12 @@ public static HiveSparkClient createHiveSparkClient(HiveConf hiveconf) throws Ex
 
     // set default spark configurations.
     sparkConf.put("spark.master", SPARK_DEFAULT_MASTER);
-    sparkConf.put("spark.app.name", SPARK_DEFAULT_APP_NAME);
+    final String appNameKey = "spark.app.name";
+    String appName = hiveConf.get(appNameKey);
+    if (appName == null) {
+      appName = SPARK_DEFAULT_APP_NAME;
+    }
+    sparkConf.put(appNameKey, appName);
     sparkConf.put("spark.serializer", SPARK_DEFAULT_SERIALIZER);
     sparkConf.put("spark.kryo.referenceTracking", SPARK_DEFAULT_REFERENCE_TRACKING);
 
-- 
1.7.9.5

